adoc/Harbor-Private-Registry-Userguide.adoc

```diff
@@ -0,0 +1,230 @@
+
+image::registry-harbor-vulnerability-scanning.png[scaledwidth=80%]
+
+To run a vulnerability scan, select the artifacts to scan and then click the Scan button.
+You can optionally select the checkbox at the top to select all artifacts in the repository.
+
+Read more about vulnerability scanning in the link:https://goharbor.io/docs/2.0.0/administration/vulnerability-scanning/scan-individual-artifact[upstream documentation].
+
+=== Image Signing
+
+By default, `Notary`, the solution for managing the content trust is installed and enabled when Harbor is installed using the Helm chart.
+This allows users to store signed images in {spr}, and in turn they have the option to use only signed images from the client applications.
+
+To use command line tools together with the `Notary` signing support in Harbor:
+
+. Enable Docker Content Trust
++
+Set these environment variables on your client:
++
+[source,bash]
+----
+export DOCKER_CONTENT_TRUST=1
+export DOCKER_CONTENT_TRUST_SERVER=https://<url_for_notary_service>
+----
++
+[NOTE]
+====
+When `DOCKER_CONTENT_TRUST` variable is set to 1, you cannot pull unsigned images from any sources.
+So for the time you want to pull some unsigned image (so you can push it signed into your local registry), it's necessary to unset the variable, or set its value to 0.
+====
+
+. Make sure your certificates are correctly installed on your system. Refer to: <<install-tls-security>>.
+
+. Push a signed Image to the private registry
++
+[source,bash]
+----
+> export DOCKER_CONTENT_TRUST_SERVER=https://notary.harbor.domain
+> export DOCKER_CONTENT_TRUST=1
+> docker push core.harbor.domain/library/alpine:latest
+The push refers to repository [core.harbor.domain/library/alpine]
+50644c29ef5a: Pushed
+latest: digest: sha256:a15790640a6690aa1730c38cf0a440e2aa44aaca9b0e8931a9f2b0d7cc90fd65 size: 528
+Signing and pushing trust metadata
+You are about to create a new root signing key passphrase. This passphrase
+will be used to protect the most sensitive key in your signing system. Please
+choose a long, complex passphrase and be careful to keep the password and the
+key file itself secure and backed up. It is highly recommended that you use a
+password manager to generate the passphrase and keep it safe. There will be no
+way to recover this key. You can find the key in your config directory.
+Enter passphrase for new root key with ID a69b97e:
+Repeat passphrase for new root key with ID a69b97e:
+Enter passphrase for new repository key with ID 5419081:
+Repeat passphrase for new repository key with ID 5419081:
+Finished initializing "core.harbor.domain/library/alpine"
+Successfully signed core.harbor.domain/library/alpine:latest
+----
++
+You will be asked for the passhprases for a new root and new repository key. The root key only need to be set once.
+The repository key *can* be changed.
+//TODO - How do you change this key?
```

Comment:
Exactly: I do not know. So unless we find out, we better remove that sentence.

(I have a feeling that it can be done with `notary` command line client, which we do **not** ship.)